lib/kubeclient/common.rb

```diff
@@ -115,7 +117,7 @@ def handle_exception
       end
       err_message = json_error_msg['message'] || e.message
       error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
-      raise error_klass.new(e.http_code, err_message, e.response)
+      raise error_klass.new((e.http_code || 404), err_message, e.response)
```

Comment:
To fix some of the rubocop errors you are getting you might want to consider a helper method for this:

```diff
-  error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
-  raise error_klass.new(e.http_code, err_message, e.response)
+  raise http_error(err_message, e.response, e.http_code)

+def http_error(err_message, e.response, http_code = 404)
+  error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
+  error_klass.new(e.http_code, err_message, e.response)
+end
```